autorest/azure/async.go

```diff
@@ -37,6 +38,80 @@ const (
 	operationSucceeded  string = "Succeeded"
 )
 
+// Future provides a mechanism to access the status and results of an asynchronous request.
+// Since futures are stateful they should be passed by value to avoid race conditions.
+type Future struct {
+	req  *http.Request
+	resp *http.Response
+	ps   pollingState
+}
+
+// NewFuture returns a new Future object initialized with the specified request.
+func NewFuture(req *http.Request) Future {
+	return Future{req: req}
+}
+
+// Response returns the last HTTP response or nil if there isn't one.
```

Comment:
What would your thoughts be on also saving a copy of the initial Response that confirmed the LRO was kicked successfully.

Reply:
That response will be saved.  The generated code will make the initial call to Done() before returning the future so Response() will return the initial response.